src/NServiceBus.AcceptanceTests/DelayedDelivery/When_Deferring_a_message.cs

```diff
@@ -13,7 +13,7 @@ public async Task Should_delay_delivery()
         {
             Requires.DelayedDelivery();
 
-            var delay = TimeSpan.FromSeconds(2);
+            var delay = TimeSpan.FromSeconds(20); // High value needed as most transports have multi second delay latency by default/
```

Comment:
This will add 18s to the test run time, is there a way to address this? Can this be made configurable for each transport?

Reply:
I've tried to do this in the past by using partials but failed. Probably away to achieve this would be to have hidden settings backdoor for this test and then create the delay as part of the When lambda where we have access to the settings.

Reply:
well, for sure in most cases 10 seconds would be sufficient. No matter which transport, 2 seconds is just too short due to the parallel execution of tests.

The *only* was would be if we could actually detect if a message was send via "delayed delivery" without looking at time. Basically, enforce transports to use the "delayed delivery logic. However, that would be a white box test that shouldn't really be part of the AT's.

Reply:
> each transport anyway should have dedicated tests for the delay infrastructure

I don't agree, why would a transport have its own AT for testing delayed delivery?

A transport should only have AT's for actually testing its own infrastructure behavior but not the actual black box high level AT of delayed sending.

Reply:
> I don't agree, why would a transport have its own AT for testing delayed delivery?

See

https://github.com/Particular/NServiceBus.AmazonSQS/tree/master/src/NServiceBus.Transport.SQS.AcceptanceTests/DelayedDelivery
https://github.com/Particular/NServiceBus.RabbitMQ/tree/master/src/NServiceBus.Transport.RabbitMQ.AcceptanceTests/DelayedDelivery
https://github.com/Particular/NServiceBus.SqlServer/tree/master/src/NServiceBus.Transport.SqlServer.AcceptanceTests/NativeTimeouts

These tests are crucial for transports. It would be a different story if we had a full test suite like we have for the outbox and saga persistence.

Of course our tests are not entirely puristics from the sense of acceptance test (user acceptance test) but they are very handy and have enabled us to find bugs and continuously make sure transports do not regress.